server/mdm/microsoft/wstep.go

```diff
@@ -0,0 +1,223 @@
+package microsoft_mdm
+
+import (
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/sha1"
```

Comment:
So I've found a way to use SHA256WithRSA certificate generation, but Windows Certificate Store requires passing the certificate thumbprint (same as SHA1 fingerprint details  - see [here](https://security.stackexchange.com/questions/14330/what-is-the-actual-value-of-a-certificate-fingerprint)). So we still have to use sha1 for this. This does not affect the security of the certificate (see [here](https://www.thesslstore.com/blog/ssl-certificate-still-sha-1-thumbprint/)) and it is mostly used as a way to locate the cert on the cert store.

Having said that, I'll suggest to add `//nolint:gosec` here to avoid the lint warning and leave the code as is - I'll add  SHA256WithRSA support on my PR